fhir-path/src/main/java/com/ibm/fhir/path/function/FHIRPathStringAbstractFunction.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * (C) Copyright IBM Corp. 2019, 2021, 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.path.function;
+
+import static com.ibm.fhir.path.util.FHIRPathUtil.empty;
+import static com.ibm.fhir.path.util.FHIRPathUtil.getSingleton;
+import static com.ibm.fhir.path.util.FHIRPathUtil.hasStringValue;
+import static com.ibm.fhir.path.util.FHIRPathUtil.isSingleton;
+
+import java.util.Collection;
+import java.util.List;
+
+import com.ibm.fhir.path.FHIRPathNode;
+import com.ibm.fhir.path.FHIRPathType;
+import com.ibm.fhir.path.evaluator.FHIRPathEvaluator.EvaluationContext;
+
+/**
+ * Abstract class which contains common apply logic for all abstract string manipulation functions
+ */
+public abstract class FHIRPathStringAbstractFunction extends FHIRPathAbstractFunction {
+    
+    /**
+     * @implSpec Common apply logic for all abstract string manipulation functions. Delegates to doApply()
+     */
+    @Override
+    public Collection<FHIRPathNode> apply(EvaluationContext evaluationContext, Collection<FHIRPathNode> context, List<Collection<FHIRPathNode>> arguments) {
+        if(context.isEmpty()) {
```

Comment:
minor style thing:  in this project we always put a space between `if` and `(` to distinguish from function calls (and same for other control flow keywords `for`, `while`, etc.)